### PR TITLE
New chart: Hubble

### PR DESCRIPTION
**Pull request by juan131**
### Description of the change

This PR adds a new chart to the catalog: [Hubble](https://github.com/cilium/hubble), a fully distributed networking and security observability platform for cloud native workloads.

This solution is compounded by 2 main components:

- Hubble Relay (deployment).
- Hubble UI (deployment). This component is also divided in 2 subcomponents running as sidecars:
  - Frontend (NGINX based container)
  - Backend (acting as a proxy between the UI (HTTP requests) and Hubble Relay (gRPC)).

Hubble Relay exposes metrics for Prometheus. We included support for the following K8s objects:

- RBAC
- HPA, VPA & PDB.
- TLS certs (both using Helm & CertManager capabilities).
- Custom configuration provided via values or existing ConfigMap (only for Relay)
- Network Policies.
- ServiceMonitor for Prometheus Operator. (only for Relay)
- Ingress resources (only for UI)

### Additional information

Similar to the Kibana-Elasticsearch model, this chart requires a running Cilium installation so Hubble Relay can connect to the Hubble Server peers.

- Related to https://github.com/bitnami/charts/issues/26443

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by juan131** [comment]
VIB tests are WIP

File: bitnami/hubble/templates/relay/cert.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  name: {{ printf "%s-crt" (include "hubble.relay.fullname" .) }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/part-of: hubble
+    app.kubernetes.io/component: relay
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  secretName: {{ printf "%s-crt" (include "hubble.relay.fullname" .) }}
```
Comment (fmulero) [inline bitnami/hubble/templates/relay/cert.yaml:19]: Shouldn't it be?

```suggestion
  secretName: {{ include "hubble.tls.relay.secretName" . }}
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Great @juan131 !

I left some comments and suggestions, could you review them?

File: bitnami/hubble/templates/relay/cert.yaml (hunk 1 above)
Comment (juan131) [reply bitnami/hubble/templates/relay/cert.yaml:19]: Optional but not mandatory, that template is returning the same value if an existing secret isn't provided.

Comment (juan131) [reply bitnami/hubble/templates/relay/cert.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/templates/client-cert.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+metadata:
+  name: {{ printf "%s-client-crt" (include "common.names.fullname" .) }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/part-of: hubble
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  secretName: {{ printf "%s-client-crt" (include "common.names.fullname" .) }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/templates/client-cert.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/templates/ui/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if not .Values.ui.autoscaling.hpa.enabled }}
+  replicas: {{ .Values.ui.replicaCount }}
+  {{- end }}
+  {{- if .Values.ui.updateStrategy }}
+  strategy: {{- toYaml .Values.ui.updateStrategy | nindent 4 }}
+  {{- end }}
+  {{- $podLabels := include "common.tplvalues.merge" (dict "values" (list .Values.ui.podLabels .Values.commonLabels) "context" .) }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/templates/ui/deployment.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/templates/ui/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.ui.enabled }}
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: Deployment
+metadata:
+  name: {{ template "hubble.ui.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/templates/ui/deployment.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/templates/ui/hpa.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    {{- if .Values.ui.autoscaling.hpa.targetMemory }}
+    - type: Resource
+      resource:
+        name: memory
+        {{- if semverCompare "<1.23-0" (include "common.capabilities.kubeVersion" .) }}
+        targetAverageUtilization: {{ .Values.ui.autoscaling.hpa.targetMemory  }}
+        {{- else }}
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.worker.autoscaling.hpa.targetMemory }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/templates/ui/hpa.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/values.yaml [hunk 6]
```diff
@@ -0,0 +1,1426 @@
...
+  extraEnvVars: []
+  ## @param relay.extraEnvVarsCM Name of existing ConfigMap containing extra env vars for Hubble Relay containers
+  ##
+  extraEnvVarsCM: ""
+  ## @param relay.extraEnvVarsSecret Name of existing Secret containing extra env vars for Hubble Relay containers
+  ##
+  extraEnvVarsSecret: ""
+  ## @param relay.extraVolumes Optionally specify extra list of additional volumes for the Hubble Relay pods
+  ##
+  extraVolumes: []
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/values.yaml:392]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/values.yaml [hunk 7]
```diff
@@ -0,0 +1,1426 @@
...
+  ingress:
+    ## @param ui.ingress.enabled Enable ingress record generation for Hubble UI
+    ##
+    enabled: false
+    ## @param ui.ingress.pathType Ingress path type
+    ##
+    pathType: ImplementationSpecific
+    ## @param ui.ingress.apiVersion Force Ingress API version (automatically detected if not set)
+    ##
+    apiVersion: ""
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/values.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/values.yaml [hunk 8]
```diff
@@ -0,0 +1,1426 @@
...
+  ## ref: https://kubernetes.io/docs/tasks/administer-cluster/configure-multiple-schedulers/
+  ##
+  schedulerName: ""
+  ## @param ui.terminationGracePeriodSeconds Seconds Hubble UI pods need to terminate gracefully
+  ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods
+  ##
+  terminationGracePeriodSeconds: ""
+  ## @param ui.extraVolumes Optionally specify extra list of additional volumes for the Hubble UI pods
+  ##
+  extraVolumes: []
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/values.yaml:1086]: done at https://github.com/bitnami/charts/pull/26579/commits/e6b5616dc0f607c5df45186f504b1588c96ed484

File: bitnami/hubble/templates/relay/networkpolicy.yaml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        {{- range .Values.relay.extraContainerPorts }}
+        - port: {{ .containerPort }}
+        {{- end }}
+      {{- if not .Values.relay.networkPolicy.allowExternal }}
+      from:
+        - podSelector:
+            matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 14 }}
+        - podSelector:
+            matchLabels:
+              {{ template "common.names.fullname" . }}-client: "true"
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/templates/relay/networkpolicy.yaml:63]: done at https://github.com/bitnami/charts/pull/26579/commits/53dba70312a17b0c77fd8f8f74662d600cefbddb

File: bitnami/hubble/templates/ui/networkpolicy.yaml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        - port: {{ .containerPort }}
+        {{- end }}
+      {{- if not .Values.ui.networkPolicy.allowExternal }}
+      from:
+        - podSelector:
+            matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 14 }}
+        - podSelector:
+            matchLabels:
+              {{ template "common.names.fullname" . }}-client: "true"
+        {{- if .Values.ui.networkPolicy.ingressNSMatchLabels }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/hubble/templates/ui/networkpolicy.yaml]: done at https://github.com/bitnami/charts/pull/26579/commits/53dba70312a17b0c77fd8f8f74662d600cefbddb

**Comment by juan131** [comment]
Closed in favor of https://github.com/bitnami/charts/pull/27526

**Pull request closed without merging**